### PR TITLE
Only process {svcname}Server interfaces in svcdef.New

### PR DESCRIPTION
**Pull request by pauln**
Also logs unexpected interfaces (i.e. oneof interfaces) so that the user can see that they were skipped.  This prevents the panic mentioned in #225, but does not add actual support for `oneof` fields.

### CODE REVIEW

**Review by adamryman — APPROVED** [review]
Pending @zaquestion or @hasLeland. Looks like it will solve the panic :+1:

File: svcdef/svcdef.go [hunk 1]
```diff
@@ -228,9 +229,15 @@ func New(goFiles map[string]io.Reader, protoFiles map[string]io.Reader) (*Svcdef
...
-				// skip the "Client" interface.
-				if strings.HasSuffix("Client", t.Name.Name) {
+				// care about, but structured a bit differently. Additionally,
+				// oneof fields generate an interface which is not a service - so
+				// for simplicity, only process the "Server" interface.
+				if !strings.HasSuffix(t.Name.Name, "Server") {
+					if !strings.HasSuffix(t.Name.Name, "Client") {
+						// This interface isn't either Server or Client; it may be a oneof
+						// field, which isn't currently supported.  Warn the user and skip.
+						log.Warnf("Unexpected interface %s found; skipping", t.Name.Name)
```
Comment (hasLeland) [inline svcdef/svcdef.go:239]: :+1: Glad to add this warning and have this explicitly ignored until one day (maybe) it can garner support.

Comment (hasAdamr) [inline svcdef/svcdef.go]: Not an approve?

Comment (adamryman) [reply svcdef/svcdef.go:239]: Not an Approve?

**Review by hasLeland — APPROVED** [review]

File: svcdef/svcdef.go (hunk 1 above)
Comment (hasLeland) [reply svcdef/svcdef.go:239]: Other things briefly caught my attention between submitting and clicking the button. Patience pls : )

**Pull request merged**